src/randomness.rs

```diff
@@ -67,7 +77,9 @@ impl RequestState {
       // if a server public key was specified, attempt to verify the
       // result of the randomness
       if let Some(pk) = public_key {
-        if !ppoprf::Client::verify(pk, blinded_point, result, self.epoch()) {
+        if result.proof.is_some()
```

Comment:
We need some way of the client enforcing verifiability if they want to. In this case, if the server returns no proof, then the verifiability check is enforced. In other words, if the public key is included, then the client needs to check that the proof exists and that it verifies correctly, otherwise it is a failure.

Reply:
added a check here: https://github.com/brave-experiments/nested-sta-rs/pull/8/files#diff-adea2445327e39eb215de697b440dbbc02135357380fb612a10a9e659129a46bR73-R77